internal/yunikorn/service.go

```diff
@@ -29,7 +29,15 @@ type Service struct {
 	workqueue *workqueue.WorkQueue
 }
 
-func NewService(ctx context.Context, repository repository.Repository, eventRepository repository.EventRepository, client Client) *Service {
+type Option func(*Service)
+
+func WithSyncInterval(interval time.Duration) Option {
+	return func(s *Service) {
+		s.syncInterval = interval
+	}
+}
+
+func NewService(repository repository.Repository, eventRepository repository.EventRepository, client Client, opts ...Option) *Service {
```

Comment:
`ctx` is deleted - do/will we need a Context for doing cancellation, etc, in the future?

Reply:
the context was actually unused, it got changed now to start a goroutine during instantiate, so it was just sitting idle

Reply:
👍🏼